### ISSUE TITLE
The UI for the Godot editor doesn't display any text.

### ISSUE THREAD

**Issue by Kinad-333** [godotengine/godot#64090]
### Godot version

3.5.1

### System information

Linux 5.18.16_1, Intel UHD Graphics 620

### Issue description

When i open Godot, the user interface doesn't display any text. Here is an image:
![image](https://user-images.githubusercontent.com/86479460/183326667-01fce918-8b7f-4367-b380-288966621f2f.png)
Occurred after the atulization of godot 3.4 to 3.5

### Steps to reproduce

error shown when running by terminal:
![image](https://user-images.githubusercontent.com/86479460/183327042-06022243-7c1a-49f5-8cde-1ef30be5adcd.png)

**Comment by akien-mga** [comment]
What Linux system do you use, and where did you get Godot from? Sounds like a distro packaging mistake (using system feeetype compiled without brotli support).

**Comment by dariemlazaro** [comment]
Works for me on Linux Mint, an ubuntu based OS.

**Comment by Kinad-333** [comment]
hi I'm using Void Linux and the godot was downloaded by the distro's own package manager

**Comment by akien-mga** [comment]
Poke @Relected who did the update for Void Linux: https://github.com/void-linux/void-packages/pull/38473

You should either build against the bundled freetype (which supports brotli), or change the Void Linux freetype package to build brotli support.

**Comment by akien-mga** [comment]
I notice I have the same issue in my Mageia package btw :)

@bruvzg Sounds like brotli support in distro packaged freetype is not very common, we might need to provide a fallback, or contact all distro packagers.

**Comment by Relected** [comment]
hello, yeah i updated the godot package, unfortunately freetype is not compiled with brotli support by the previous maintainer of the package, and the last update for freetype was about 18~ hours ago by @Duncaen which might mean that he knows whats up

**Comment by bruvzg** [comment]
We can have two versions of the fonts in the repo. or use external tools to compress/decompress it, but I'm not sure how to detect which fonts to bundle, it's easy to do in the code, but not from the Scons.

For 4.0 it's also possible to fall back to system font (if fontconfig is installed).

**Comment by Relected** [comment]
update: Duncan just made a PR updating freetype with brotli support, this issue should be fixed when the PR gets merged (dont forget to update your repo after :) )

**Comment by cunyx** [comment]
Struggling with this as well since 3.5, the bundled freetype
seems to depend somehow on build-in libpng and zlib.

Unbundling libpng and zlib while using freetype from godot throws build errors for me.

**Comment by akien-mga** [comment]
Indeed, the bundled freetype is not wired to build with system zlib and libpng. It could in the past and that caused issues when the system libpng version doesn't match what freetype was built again (typically building against system libpng16 and then running on older Ubuntu - and up until recently the Steam runtime - with libpng12). And zlib is a core component for libpng.

So the recommended option is either to build freetype with brotli support (just did so for Mageia's freetype package, also added the optional harfbuzz and librsvg support while at it), or build with builtin freetype + libpng + zlib.

**Comment by Relected** [comment]
yeah it fully fixed now, this issue should be closed

**Comment by akien-mga** [comment]
I'm keeping it open for a few days for discoverability by other distro packagers. But thanks for solving it promptly for Void Linux :+1:

**Comment by cemkeylan** [comment]
I'm a bit confused, I noticed this while compiling godot and I randomly caught the build system saying built-in fonnts are disabled because freetype is built without brotli support. I can't really make sense of it, is there a way to set system fonts without having to launch the editor? If there is no way to display text without freetype with brotli, why does the build system let you compile it in the first place?

**Comment by Calinou** [comment]
> I can't really make sense of it, is there a way to set system fonts without having to launch the editor

4.x will fall back to system fonts in this case. However, 3.x can't load system fonts as that feature wasn't implemented there (and won't be backported due to how much work it requires).
